modules/test/playwright/pages/object-web/object-layout/ObjectLayoutsPage.ts

```diff
@@ -0,0 +1,82 @@
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+import {FrameLocator, Locator, Page} from '@playwright/test';
+
+import {ViewObjectDefinitionsPage} from '../ViewObjectDefinitionsPage';
+
+export class ObjectLayoutsPage {
+	readonly addBlock: Locator;
+	readonly addField: Locator;
+	readonly addObjectLayoutButton: Locator;
+	readonly addTab: Locator;
+	readonly iframeLocator: FrameLocator;
+	readonly layoutNameBox: Locator;
+	readonly layoutTab: Locator;
+	readonly layoutsTabItem: Locator;
+	readonly page: Page;
+	readonly saveBlockButton: Locator;
+	readonly saveLayoutButton: Locator;
+	readonly saveTabButton: Locator;
+	readonly tabAndBlockNameBox: Locator;
+	readonly viewObjectDefinitionsPage: ViewObjectDefinitionsPage;
+
+	constructor(page: Page) {
+		this.iframeLocator = page.frameLocator('iframe');
+		this.addBlock = this.iframeLocator.getByRole('button', {
+			name: 'Add Block',
+		});
+		this.addField = this.iframeLocator.getByRole('button', {
+			name: 'Add Field',
+		});
+		this.addObjectLayoutButton = page.getByLabel('Add Object Layout');
+		this.addTab = this.iframeLocator.getByRole('button', {name: 'Add Tab'});
+		this.layoutNameBox = page.getByLabel('Name');
+		this.layoutTab = this.iframeLocator.getByRole('tab', {name: 'Layout'});
+		this.layoutsTabItem = page.getByRole('link', {name: 'Layouts'});
+		this.page = page;
+		this.saveBlockButton = this.iframeLocator
+			.getByLabel('Add Block')
+			.getByRole('button', {name: 'Save'});
+		this.saveLayoutButton = page.getByRole('button', {name: 'Save'});
+		this.saveTabButton = this.iframeLocator
+			.getByLabel('Add Tab')
+			.getByRole('button', {name: 'Save'});
+		this.tabAndBlockNameBox = this.iframeLocator.getByLabel('Label');
+		this.viewObjectDefinitionsPage = new ViewObjectDefinitionsPage(page);
+	}
+
+	async createObjectLayout(objectLayoutName: string) {
+		await this.addObjectLayoutButton.click();
+		await this.layoutNameBox.click();
+		await this.layoutNameBox.fill(objectLayoutName);
+		await this.saveLayoutButton.click();
+	}
+
+	async goto(objectDefinitionLabel: string) {
+		await this.viewObjectDefinitionsPage.goto();
+
+		await this.viewObjectDefinitionsPage.clickEditObjectDefinitionFDSLink(
+			objectDefinitionLabel
+		);
+
+		await this.layoutsTabItem.click();
+	}
+
+	async openObjectLayout(objectLayoutName: string) {
+		await this.page.getByRole('link', {name: objectLayoutName}).click();
+		await this.layoutTab.click();
+		await this.addTab.click();
+		await this.tabAndBlockNameBox.click();
+		await this.tabAndBlockNameBox.fill(objectLayoutName);
+		await this.saveTabButton.click();
+		await this.addBlock.click();
+		await this.tabAndBlockNameBox.click();
+		await this.tabAndBlockNameBox.fill(objectLayoutName);
+		await this.saveBlockButton.click();
+		await this.addField.click();
+		await this.iframeLocator.getByText('Select an Option').click();
+	}
```

Comment:
Hello @vngoliveira, this function performs many actions and will be less reusable. We can try to break this function into more smaller functions to improve reusability. For example:

```
	async openObjectLayoutConfiguration(objectLayoutName: string) {
		await this.page.getByRole('link', {name: objectLayoutName}).click();
		await this.layoutTab.click();
	}

	async createObjectLayoutBlock(objectLayoutBlockName: string) {
		await this.addTab.click();
		await this.labelInput.fill(objectLayoutBlockName);
		await this.saveTabButton.click();
	}

	async createObjectLayoutTab(objectLayoutTabName: string) {
		await this.addTab.click();
		await this.labelInput.fill(objectLayoutTabName);
		await this.saveTabButton.click();
	}
```